docs/apicast-crd-reference.md

```diff
@@ -0,0 +1,210 @@
+        memory: "128Mi"
+
+  production:
+    externalDnsHostname: "apicast-production.test.stg-saas.3sca.net"
+    image:
+      name: quay.io/3scale/apicast-cloud-hosted
+      tag: apicast-v3.8.0-r3
+      pullSecretName: quay-pull-secret
+    replicas: 1
+    env:
+      configurationCache: "300"
+      threescalePortalEndpoint: "http://mapping-service/config"
+      apicastLogLevel: debug
+      apicsatOIDCLogLevel: debug
+    marin3r:
+      enabled: true
+      annotations:
+        marin3r.3scale.net/node-id: apicast-staging
+        marin3r.3scale.net/ports: gateway-http:38080,gateway-https:38443,envoy-metrics:9901
+    livenessProbe:
+      initialDelaySeconds: 25
+      timeoutSeconds: 2
+      periodSeconds: 20
+      successThreshold: 1
+      failureThreshold: 5
+    readinessProbe:
+      initialDelaySeconds: 25
+      timeoutSeconds: 2
+      periodSeconds: 20
+      successThreshold: 1
+      failureThreshold: 5
+    resources:
+      requests:
+        cpu: 500m"
+        memory: "64Mi"
+      limits:
+        cpu: "1"
+        memory: "128Mi"
+
+  loadBalancer:
+    proxyProtocol: "*"
+    crossZoneLoadBalancingEnabled: true
+    connectionDrainingEnabled: true
+    connectionDrainingTimeout: 60
+    connectionHealthcheckHealthyThreshold: 2
+    connectionHealthcheckUnhealthyThreshold: 2
+    connectionHealthcheckInterval: 5
+    connectionHealthcheckTimeout: 3
+
+  grafanaDashboard:
+    label:
+      key: discovery
+      value: enabled
+```
+
+## CR Spec
+
+|                             **Field**                             | **Type**  | **Required** |           **Default value**           |                             **Description**                              |
+| :---------------------------------------------------------------: | :-------: | :----------: | :-----------------------------------: | :----------------------------------------------------------------------: |
+|                       `staging.image.name`                        | `string`  |      No      | `quay.io/3scale/apicast-cloud-hosted` |                      Image name (docker repository)                      |
+|                        `staging.image.tag`                        | `string`  |     Yes      |                   -                   |                                Image tag                                 |
+|                  `staging.image.pullSecretName`                   | `string`  |      No      |                   -                   |                 Quay pull secret for private repository                  |
+|                        `staging.replicas`                         |   `int`   |      No      |                  `1`                  |                            Number of replicas                            |
+|              `staging.env.apicastConfigurationCache`              | `string`  |     Yes      |                   -                   |                     Apicast configurations cache TTL                     |
```

Comment:
Being a field representing time in seconds, shouldnt it be an integer, instead of a string? (applies to examples, docs, CRD)

Reply:
Validations added in 1d9096289bf609153e378ba6ae025774854cbc4f